infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespaceFactory.java

```diff
@@ -46,23 +56,26 @@
  */
 @Singleton
 public class KubernetesNamespaceFactory {
+  private static final Logger LOG = LoggerFactory.getLogger(KubernetesNamespaceFactory.class);
 
-  private static final Map<String, Function<Subject, String>> NAMESPACE_NAME_PLACEHOLDERS =
-      new HashMap<>();
+  private static final Map<String, Function<PlaceholderResolutionContext, String>>
+      NAMESPACE_NAME_PLACEHOLDERS = new HashMap<>();
 
   static {
-    NAMESPACE_NAME_PLACEHOLDERS.put("<username>", Subject::getUserName);
-    NAMESPACE_NAME_PLACEHOLDERS.put("<userid>", Subject::getUserId);
+    NAMESPACE_NAME_PLACEHOLDERS.put("<username>", ctx -> ctx.user.getUserName());
+    NAMESPACE_NAME_PLACEHOLDERS.put("<userid>", ctx -> ctx.user.getUserId());
+    NAMESPACE_NAME_PLACEHOLDERS.put("<workspaceid>", ctx -> ctx.workspaceId);
```

Comment:
Please add this description for this placefolder into docs of `default_namespace` property in che.properties.

Reply:
Seems this PR also fixes https://github.com/eclipse/che/issues/14585